shared/src/test/diff/nu/Annotations.mls

```diff
@@ -40,6 +40,40 @@ Foo
 //│ res
 //│     = 5
 
+:e
+let x = 1
+@x 2
+//│ ╔══[ERROR] Type mismatch in annotated integer literal:
+//│ ║  l.45: 	@x 2
+//│ ║        	 ^^^
+//│ ╟── integer literal of type `1` is not an instance of type `Annotation`
+//│ ║  l.44: 	let x = 1
+//│ ║        	        ^
+//│ ╟── but it flows into reference with expected type `Annotation`
+//│ ║  l.45: 	@x 2
+//│ ╙──      	 ^
+//│ let x: 1
+//│ 2
+//│ x
+//│   = 1
+//│ res
+//│     = 2
+
+:e
+let x = 1
+@x
+fun foo(x) = 1
+//│ ╔══[ERROR] Type mismatch in definition:
+//│ ║  l.65: 	fun foo(x) = 1
+//│ ║        	    ^^^^^^^^^^
+//│ ╟── integer literal of type `1` is not an instance of type `Annotation`
+//│ ║  l.63: 	let x = 1
+//│ ╙──      	        ^
+//│ let x: 1
+//│ fun foo: anything -> 1
+//│ x
+//│   = 1
+
```

Comment:
Where are these new errors coming from?

Reply:
It's because `@tailrec` and `@tailcall` can only be used on functions and function calls respectively. However, the type system and grammar allow annotations to be applied to any expression, typedef or function.

I suppose the indicated location of the error could be more helpful though

Reply:
OK I see. Thanks.